### DISCUSSION TITLE
StreamingResponse streams data to random clients.

### DISCUSSION THREAD

**Discussion by kielerrr** [fastapi/fastapi#11405] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
log_messages = []

def log_message_add(game_id: int, *messages: str):
    """
    Adds a timestamped log message with a game ID to the shared log messages list.

    Args:
        game_id (int): The ID of the game.
        messages (str): The log messages to add.
    """
    for message in messages:
        log_messages.append({"gameId": game_id, "message": f"{datetime.now().strftime('%H:%M:%S')}: {message}"})
        
        
async def log_streamer():
    while True:
        if log_messages:
            log_entry = log_messages.pop(0)
            yield f"data: {json.dumps(log_entry)}\n\n"
        else:
            await asyncio.sleep(1)


@app.get("/logs")
async def get_logs():
    return StreamingResponse(log_streamer(), media_type="text/event-stream")
```

### Description

Strangely, generating logs on one client, will cause the StreamingResponse to be sent to another client instead. 
I can hit an endpoint to generate logs on my mobile session, but it streams the data to my desktop session.

I think it may be related to the log_messages list being global, but even then it seems like an odd behavior. 
I made a 7 second [video](https://youtu.be/tZEQcJfwb-M) showing the behavior. 

I will probably use another solution that StreamingResponse, but I'm curious how this behaviour can happen.
The frontend is a Vue3 app

### Operating System

Linux

### Operating System Details

Ubuntu 20 / Android

### FastAPI Version

0.109.0

### Pydantic Version

2.5.1

### Python Version

Python 3.10.13

### Additional Context

The frontend is Vue3. FastAPI runs in Kubernetes with a Traefik reverse proxy.

**Comment by jgould22** [comment]
> I think it may be related to the log_messages list being global, but even then it seems like an odd behavior. 

Why does this seem like odd behaviour?

**Comment by kielerrr** [comment]
> Why does this seem like odd behaviour?

The stream is sent to other clients, but not to the client initiating the stream.

**Comment by YuriiMotov** [comment, marked as answer]
Since your `log_messages` is global, it will be sent to any one of currently active requests.

To make it working as you want you probably need to turn `log_messages` into a dict where the key is `client_id` and the value is the list of log messages. This way you will separate logs of different clients
